h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderHelper.java

```diff
@@ -82,6 +73,90 @@ static double computePriorMean(Frame encodings, int targetClass) {
     }
   }
 
+  /**
+   * @return the index of the interaction column for the group, or the index of the column if the group has only one.
+   */
+  static int createFeatureInteraction(Frame fr, String[] colGroup) {
```

Comment:
Knit: Using Optional arguments and one method would be useful.

Reply:
Not sure I understand what you mean: is it
```
static int createFeatureInteraction(Frame fr, String[] colGroup, String... interactionDomain)
```
If this is what you mean, then I really don't like it here because we expect the `interactionDomain` array being of same length as `colGroup`, and the signature above makes them look totally independant.